platform/v1alpha1/types.go

```diff
@@ -0,0 +1,82 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Package struct {
+	// Name contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle. In
+	// the current implementation, it will be retrieving this name from a list of
+	// supported operators out of the redhat-operators catalog included with
+	// OpenShift.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern:=[a-z0-9]([-a-z0-9]*[a-z0-9])?
+	// +kubebuilder:validation:MaxLength:=56
+	Name string `json:"name"`
+}
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// Package contains the desired package and its configuration for this
+	// PlatformOperator.
+	// +kubebuilder:validation:Required
+	Package Package `json:"package"`
```

Comment:
nit: I think this might be better if it's a `*Package`? Then we could simply check for nil in the controller code, versus an empty struct.

Reply:
Oh, yeah good call.

Reply:
Thinking about it, would we be able to have a PO created with this field since it is validated as required? I don't think there is a huge deal in changing this to *Package but I mainly ask if we should be checking for it be empty in the code.

Reply:
Yes, I guess if the Package and Name fields are both required, we can rely on validation to ensure there is something there. No need to update this then I suppose. We can always revisit.

Reply:
Note as well that this is a configuration API, so [pointers](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#pointers) are discouraged